smithy-model/src/main/java/software/amazon/smithy/model/neighbor/NeighborVisitor.java

```diff
@@ -71,6 +71,10 @@
         for (ShapeId resource : shape.getResources()) {
             addBinding(result, shape, resource, RelationshipType.RESOURCE);
         }
+        // Add ERROR relationships from service -> errors.
+        for (ShapeId errorId : shape.getErrors()) {
```

Comment:
Is there a way in the selector to get all errors for an operation bound to a service? Or do we just have to join them manually

Reply:
You can do that using something like:

```
service $serviceErrors(-[error]->) ~> operation :is(${serviceErrors}, -[error]->)
```

1. Find each service
2. Store the errors of the service in a variable named `$serviceErrors`
3. Traverse every shape in the service
4. Only look at operations within the service
5. Emit every `$serviceError` plus every error attached to the operation